File: dc_plc/dc_plc/report/dc_product_rnd_specialist_stats/dc_product_rnd_specialist_stats.py
# Copyright (c) 2013, igrekus and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

from dc_plc.custom.utils import add_translation, add_completeness, add_query_relevance
from dc_plc.controllers.stats_query import get_rnd_spec_stats


def execute(filters=None):
	columns = get_columns()
	data = get_data(filters)

	return columns, data


def get_columns():
	return [
		"ID:Link/DC_PLC_Product_Summary",
		_("Relevance"),
		_("Progress"),
		_("Status"),
		_("RnD Title"),
		_("Model"),
		_("Function"),
		_("External number"),
		_("Internal number"),
	]


def get_data(filters):
	res = get_rnd_spec_stats(filters)

	# TODO use frappe API to determine perms
	has_perms = 'DC_PLC_RnD_Specialist' in frappe.get_roles(frappe.session.user)

	res = [add_completeness(row, [2]) for row in res]
	res = [add_query_relevance(row, has_perms) for row in res]

	return res

File: dc_plc/dc_plc/report/dc_product_rnd_specialist_stats/dc_product_rnd_specialist_stats.js
// Copyright (c) 2016, igrekus and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["DC Product RND Specialist Stats"] = {
	filters: [],
	formatter: frappe.dc_plc.utils.formatters.rnd_spec_formatter,
	after_datatable_render: table_instance => {
		let highlight_cols = [5];
		let data = table_instance.datamanager.data;

		for (let row = 0; row < data.length; ++row) {
			const [date, check, perms] = data[row][__('Relevance')].split(';');
			const is_checked = !!parseInt(check);
			highlight_cols.forEach(col => {
				let bgCol = is_checked ? 'rgba(37,220,2,0.2);' : 'rgba(255, 252, 29, 0.27);';
				table_instance.style.setStyle(`.dt-cell--${col}-${row}`, {backgroundColor: `${bgCol}`});
			});
		}
		table_instance.style.setStyle(`.dt-scrollable`, {height: '550px;'});
	},
	onload: report => {
		report.export_products = [];
		report.page.set_title_sub('');
		report.page.clear_actions_menu();
	}
};
